pages/sites/[id]/edit.tsx

```diff
@@ -0,0 +1,63 @@
+import { getServerSession } from "next-auth"
+import { authOptions } from "@/pages/api/auth/[...nextauth]"
+import { prisma } from "@/lib/prisma"
+import { GetServerSidePropsContext } from "next"
+import { useState } from "react"
+import { useRouter } from "next/router"
+export async function getServerSideProps(ctx: GetServerSidePropsContext) {
+    const session = await getServerSession(ctx.req, ctx.res, authOptions)
+    const id = Number(ctx.params?.id)
+    if (!session?.user?.email) {
+        return { redirect: { destination: "/", permanent: false } }
+    }
+    const site = await prisma.site.findUnique({
```

Comment:
This gives 404 error page, because site.owner.email is undefined. Change code to:
```
const site = await prisma.site.findUnique({
        where: { id },
        include: { owner: true },
    })
```